doc/y_cable/design_doc.md

```diff
@@ -0,0 +1,247 @@
+
+
+### Vendor to Module mapping
+
+#### Background
+
+- We need to have a vendor to corresponding Y cable module mappings defined and accessible somewhere for PMon container to import
+
+  - Before calling any Y cable API, how do we know which type/vendor does the Y cable belong to ?
+  - Also lets suppose once the type/vendor of the cable is known from where does PMon container or cli import the module ?
+  - Since there can be many specs which vendors follow (8636, 8624 etc.) for transceivers it is important to have a solution which meets all requirements so that the API access is available to whoever wants to invoke it.
+  - another issue is how do we support the multiple vendors modules/packages which could be used by both SONiC PMon container docker as well as the sonic cli (sonic-utilities). Basically the package should be available both in host and container. Although we are moving towards a model where cli should not need to import Y cable modules but it is still preferred have it accessible
+
+#### Proposed Solution
+
+- We define a file which would contain vendor to module mapping and put it in a place which is accessible to both PMon container and host
+- The vendor to module mapping can be present in a file which could be present in sonic_platform_common
+- It makes sense to keep the mapping file in the sonic_y_cable package so that it can be updated in the same pull request when a vendor adds a new cable implementation. However, we cannot install data files outside of the Python directory via a wheel. Considering this we propose to make this a Python file which simply contains a 2D dictionary which can be used to look up the module path. If the file is part of the sonic_y_cable package, it will be installed in both the PMon container and the host
+
+For example
+```
+	/sonic_platform_common/sonic_y_cable/y_cable_vendor_mapping.py
+```
+
+Vendors can have several implementations/ways to use this concept
+- Vendor has a 1:1 mapping between model and Python file
+- Vendor has one Python file which supports all models
+- A combination of the above
+
+- Mapping could be such that its a dictionary in 2D
+
+
+```
+	{<vendor_name_1>: {<model_name: <module>},
+	<vendor_name_2> : {<model_name>:<module>}}
+```
+- For example
+```
+	{"Credo" : {"model1": "y_cable.py"},
+ 	 "Amphenol" : {"model_a": "y_cable_a.py", "model_b":"y_cable_b.py"}}
+```
+
+#### Rationale
+
+  - This is because both the container and host can easily access this path. And it would be easy to mount this on container (PMon container) as applicable.
+  - The sonic_y_cable package is installed in both host and pmon
+
+#### Implementation details
+- Once the xcvrd will start (daemon launched) it will know the Vendor name from the register specification and then it can appropriately read and parse the vendor name from the spec. Once it has this information it can then using this file to Load/import the appropriate module.
```

Comment:
When xcvrd starts or a new transceiver is connected, it will obtain both the vendor name and part number from the EEPROM. Please include mention of part number here, as well.

Reply:
done